server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2460,6 +2460,15 @@ public SeqNoStats getSeqNoStats(long globalCheckpoint) {
         return localCheckpointTracker.getStats(globalCheckpoint);
     }
 
+    @Override
+    public void checkGlobalCheckpointBeforeClose(final long globalCheckpoint) {
+        final long maxSeqNo = getSeqNoStats(globalCheckpoint).getMaxSeqNo();
```

Comment:
I would move this implementation to the Engine class, and have specific engines explicitly opt-out of this behavior by overriding with empty method